awshelper/config_test.go

```diff
@@ -36,9 +38,61 @@ func TestAwsIsAddedInUserAgent(t *testing.T) {
 func TestAwsSessionValidationFail(t *testing.T) {
 	t.Parallel()
 
-	err := awshelper.ValidateAwsSession(&awshelper.AwsSessionConfig{
+	session, err := awshelper.CreateAwsSession(&awshelper.AwsSessionConfig{
 		Region:        "not-existing-region",
 		CredsFilename: "/tmp/not-existing-file",
 	}, options.NewTerragruntOptions())
+	require.NoError(t, err)
+
+	err = awshelper.ValidateAwsSession(session)
 	assert.Error(t, err)
 }
+
+// Test to validate cases when is not possible to read all S3 configurations
+// https://github.com/gruntwork-io/terragrunt/issues/2109
+func TestAwsNegativePublicAccessResponse(t *testing.T) {
+	t.Parallel()
+	testCases := []struct {
+		name     string
+		response *s3.GetPublicAccessBlockOutput
+	}{
+		{
+			name: "nil-response",
+			response: &s3.GetPublicAccessBlockOutput{
+				PublicAccessBlockConfiguration: nil,
+			},
+		},
+		{
+			name: "legacy-bucket",
+			response: &s3.GetPublicAccessBlockOutput{
+				PublicAccessBlockConfiguration: &s3.PublicAccessBlockConfiguration{
+					BlockPublicAcls:       nil,
+					BlockPublicPolicy:     nil,
+					IgnorePublicAcls:      nil,
+					RestrictPublicBuckets: nil,
+				},
+			},
+		},
+		{
+			name: "false-response",
+			response: &s3.GetPublicAccessBlockOutput{
+				PublicAccessBlockConfiguration: &s3.PublicAccessBlockConfiguration{
+					BlockPublicAcls:       aws.Bool(false),
+					BlockPublicPolicy:     aws.Bool(false),
+					IgnorePublicAcls:      aws.Bool(false),
+					RestrictPublicBuckets: aws.Bool(false),
+				},
+			},
+		},
+	}
+	for _, testCase := range testCases {
+		testCase := testCase
```

Comment:
NIT: We don't need this anymore in modern golang.

Reply:
Not my code. To get rid of it once and for all, I searched and deleted it in all the code.